Source/GUI/MemWatcher/MemWatchWidget.cpp

```diff
@@ -273,6 +273,43 @@ void MemWatchWidget::onMemWatchContextMenuRequested(const QPoint& pos)
   connect(copy, &QAction::triggered, this, [this] { copySelectedWatchesToClipBoard(); });
   contextMenu->addAction(copy);
 
+  if (index != QModelIndex())
+  {
+    MemWatchEntry* entry = m_watchModel->getEntryFromIndex(index);
+    if (entry->isBoundToPointer())
+    {
+      QMenu* copyAddrSubmenu = contextMenu->addMenu(tr("Copy add&ress..."));
+      QAction* copyPointer = new QAction(tr("Copy &base address..."), this);
+      connect(copyPointer, &QAction::triggered, this,
+              [this, entry] { copyAddressToClipboard(entry->getConsoleAddress()); });
+      copyAddrSubmenu->addAction(copyPointer);
+      for (int i = 0; i < static_cast<int>(entry->getPointerLevel()); ++i)
+      {
+        std::string strAddressOfPath = entry->getAddressStringForPointerLevel(i + 1);
+        if (strAddressOfPath == "???")
+          break;
+        QAction* showAddressOfPathInViewer = new QAction(
+            tr("Copy pointed address at &level %1...").arg(QString::number(i + 1)), this);
+        connect(showAddressOfPathInViewer, &QAction::triggered, this,
+                [this, entry, i] { copyAddressToClipboard(entry->getAddressForPointerLevel(i + 1)); });
+        copyAddrSubmenu->addAction(showAddressOfPathInViewer);
+      }
+    }
+    else
+    {
+      QAction* copyPointer = new QAction(tr("Copy add&ress"), this);
+      connect(copyPointer, &QAction::triggered, this,
+              [this, entry] { copyAddressToClipboard(entry->getConsoleAddress()); });
+      contextMenu->addAction(copyPointer);
+
+      QModelIndexList selection = m_watchView->selectionModel()->selectedRows();
+      if (selection.count() == 0)
+      {
+        copyPointer->setEnabled(false);
+      }
```

Comment:
I realized that since it checks for a valid index to get to this scope, there is no way the selection would have zero entries, so I removed this check altogether